src/components/NotificationStack.tsx

```diff
@@ -249,17 +250,43 @@ export default function NotificationStack() {
     <GestureHandlerRootView style={styles.container}>
       <GestureDetector gesture={Gesture.Exclusive(flingUp)}>
         <View style={styles.viewWrapper}>
-          {(uaiStack || []).map((c, index) => {
-            return (
-              <Card
-                info={c}
-                key={c.id}
-                index={index}
-                totalLength={uaiStack.length - 1}
-                activeIndex={activeIndex}
-              />
-            );
-          })}
+          {uaiStack.length ? (
+            (uaiStack || []).map((c, index) => {
+              return (
+                <Card
+                  info={c}
+                  key={c.id}
+                  index={index}
+                  totalLength={uaiStack.length - 1}
+                  activeIndex={activeIndex}
+                />
+              );
+            })
+          ) : (
+            <Box
```

Comment:
Please make a component for this. Avoid hardcoding component defs in parent component. Makes the code un-readable.